### ISSUE TITLE
How to pipe a stream into the ssh shell?

### ISSUE THREAD

**Issue by kentnek** [mscdex/ssh2#690]
Hi, I'm trying to write a `ssh2` client that logs into a server, does a few things, then uploads a file to that server via Zmodem (can't use sftp/scp here). Essentially I want to convert this Python [gist](https://github.com/ThomasHabets/ssh-scripts/blob/master/ssmodem) into Nodejs.

This is what I have so far:
```js
conn.shell(function(err, stream) {
    ...
    stream.write("rz -e\n"); // begin to receive file on the server
    const child = spawn("sz", ["-e", "test.txt"]); // spawn "sz" locally to send file
    
    // Pipe child's stdio to the shell's stream
    child.stdin.pipe(stream);
    child.stdout.pipe(stream);

    ...
});
```

However, the code does not work. Setting the child's `stdio` property fails to work as well:
```js
const child = spawn("sz", ["-e", "test.txt"], { stdio: [stream, stream, process.stderr] }); 
```

How should I pass the input/output streams of the child to the shell's stream? Thanks in advance!

**Comment by mscdex** [comment]
Two things I see right away:

1. Consider using `conn.exec()` instead of `conn.shell()`
2. `child.stdin` is not a Readable stream, it's a Writable stream, so you can't `pipe()` it anywhere
3. You're not piping `stream` back to `child.stdin` (e.g. `stream.pipe(child.stdin)`)

**Comment by kentnek** [comment]
Wow that's a fast reply! I was able to solve it myself and about to update this issue haha.

So upon further reading, I also found that I had to `pipe` a `Readable` stream to a `Writable` one instead. This managed to work:
```js
stream.stdout.pipe(child.stdin);
child.stdout.pipe(stream.stdin, { end: false }); // prevent stream.stdin from closing because of child
```
Regarding `conn.shell()`, I actually need to perform another `ssh` and `su` from the bastion server to access another server, before calling `rz`.
